src/equations/equations.jl

```diff
@@ -329,5 +333,14 @@ include("lattice_boltzmann_3d.jl")
 abstract type AbstractAcousticPerturbationEquations{NDIMS, NVARS} <: AbstractEquations{NDIMS, NVARS} end
 include("acoustic_perturbation_2d.jl")
 
+abstract type AbstractParabolicEquations{NDIMS, NVARS} <: AbstractEquations{NDIMS, NVARS} end
+
+# operator types used for dispatch on parabolic boundary fluxes
+struct Gradient end
+struct Divergence end
```

Comment:
I think the `Gradient`/`Divergence` part should get some more documentation on what the concept is behind it and how to correctly use them. As far as I understand, they are needed since in the `DGMulti` solver, BCs are separated into BCs applied in two parts of the spatial discretization. This allows each BC to provide some parts in the form of a flux, some parts in the form of an "outer" state. Is my understanding correct? However, why are there two separate phases, i.e., why isn't there just "one" BC call that does The Right Thing®️, either flux-based or state based? Or is this a different approach to applying BCs than we use in the TreeMesh-related solvers?

This is an honest question, I think I just don't fully get the whole idea. If it is too complicated to explain in text, maybe we can also discuss it on Tuesday.

Reply:
I can add some comments here, but I've added a box for more complete documentation in https://github.com/trixi-framework/Trixi.jl/issues/1147.

Reply:
Sounds good. Maybe we can discuss it a little at the Trixi meeting such that I get a better idea.

Reply:
My understanding of parabolic DG is that every time you take a derivative of `u`, you impose BCs on `u` (and similarly for divergence of `u` and BCs on `u·n`). Since parabolic DG takes a gradient of `u` (call this `sigma`) and then divergence of `sigma`, we can impose BCs on both `u` and `sigma`. I'm specifying both of those explicitly.

That said, one could use `boundary_condition_do_nothing` as a default. For advection-diffusion, this does The Right Thing®️. I'm not 100% sure if this generalizes to more complex equations, but I think it might...

Reply:
I've added "default" BCs in https://github.com/trixi-framework/Trixi.jl/pull/1136/commits/4c3a7edfafc1becc07bd74e396f33bd255632c91. This way, users don't have to specify both `Gradient` and `Divergence` boundary conditions, but they still need to specify whether the boundary condition (e.g., Dirichlet or Neumann) corresponds to the `Gradient` or `Divergence`.

Reply:
I am not sure about allowing a default behavior is good practice in terms of POLA (principle of least astonishment). As discussed before, maybe we should force users to explicitly give a "do nothing" BC for those cases. But this can be discussed later - right now we are still (supposed to be) inside the "make it work" phase :-)

Reply:
Sounds good. I've added this as a discussion item in https://github.com/trixi-framework/Trixi.jl/issues/1147